tests/test_transforms/test_math/test_sklearn_transform_interface.py

```diff
@@ -278,3 +278,319 @@ def test_ordering(transform_constructor, in_column, mode, multicolumn_ts):
         df_multi = transformed_df.loc[:, pd.IndexSlice[segments, column_multi]]
         df_single = transformed_dfs_one_column[i].loc[:, pd.IndexSlice[segments, column_single]]
         assert np.all(df_multi == df_single)
+
+
+@pytest.mark.parametrize("inplace", [False, True])
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "mode",
+    [
+        "macro",
+        "per-segment",
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_transform_not_fitted_fail(transform_constructor, mode, in_column, inplace, multicolumn_ts):
+    df = multicolumn_ts.to_pandas()
+    transform = transform_constructor(mode=mode, in_column=in_column, inplace=inplace)
+
+    with pytest.raises(ValueError, match="The transform isn't fitted"):
+        _ = transform.transform(df)
+
+
+@pytest.mark.parametrize("inplace", [False, True])
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "mode",
+    [
+        "macro",
+        "per-segment",
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_inverse_transform_not_fitted_fail(transform_constructor, mode, in_column, inplace, multicolumn_ts):
+    df = multicolumn_ts.to_pandas()
+    transform = transform_constructor(mode=mode, in_column=in_column, inplace=inplace)
+
+    with pytest.raises(ValueError, match="The transform isn't fitted"):
+        _ = transform.inverse_transform(df)
+
+
+@pytest.mark.parametrize("inplace", [False, True])
+@pytest.mark.parametrize(
+    "in_column",
+    [
+        "exog_1",
+        ["exog_1", "exog_2"],
+    ],
+)
+@pytest.mark.parametrize(
+    "mode",
+    [
+        "macro",
+        "per-segment",
+    ],
+)
+@pytest.mark.parametrize(
+    "transform_constructor",
+    [
+        BoxCoxTransform,
+        YeoJohnsonTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+        MaxAbsScalerTransform,
+        StandardScalerTransform,
+        RobustScalerTransform,
+        MinMaxScalerTransform,
+    ],
+)
+def test_transform_subset_segments(transform_constructor, mode, in_column, inplace, multicolumn_ts):
```

Comment:
Can we add a test to ensure that additional columns (columns from `df_to_add`) are not leaked into the final results for per-segment mode?

Reply:
I don't really understand how are we going to check this.

Reply:
Sure, just to check that `test_df` and transformed version of `test_df` have the same set of segments. Since we add and remove columns in the middle of transformation.